docs/state-of-rent.md

```diff
@@ -0,0 +1,119 @@
+- Fixed an issue on testnet, mainnet was unaffected
+
+#### `prevent_rent_paying_rent_recipients`
+Feature address: https://explorer.solana.com/address/Fab5oP3DmsLYCiQZXdjyqT3ukFFPrsmqhXU4WU1AWVVF
+Feature issue: https://github.com/solana-labs/solana/issues/30151
+Activated on mainnet in epoch 542 (~2023-12-05)
+- Does not allow rent rewards to be paid to a validator account if it would make that account rent-paying
+
+#### `set_exempt_rent_epoch_max`
+Feature address: https://explorer.solana.com/address/5wAGiy15X1Jb2hkHnPDCM8oB9V42VNA9ftNVFK84dEgv
+Feature issue: https://github.com/solana-labs/solana/issues/28683
+Activated on mainnet in epoch 570 (~2024-02-05)
+- Initializes new accounts with a rent epoch set to the marker value (`RENT_EXEMPT_RENT_EPOCH`)
+- During rent collection, sets the rent epoch for "rent-exempt" accounts to the `RENT_EXEMPT_RENT_EPOCH` marker as well
+- This has the side effect of not needing to update accounts every epoch anymore.
+- Note that this feature behavior didn't remove accounts from account delta hash
+calculations because when we run rent collection for an account, even if there
+are no changes made, the account is still considered "written" in that block.
+This behavior of writing an account without any actual changes is called a
+"rewrite" and the `skip_rent_rewrites` feature aims to remove rewrites.
+
+#### `validatee_fee_collector_account`
+Feature address: https://explorer.solana.com/address/prpFrMtgNmzaNzkPJg9o753fVvbHKqNrNTm76foJ2wm
+Feature issue: https://github.com/solana-labs/solana/issues/33888
+Activated on mainnet in epoch 598 (~2024-04-04)
+- Prevents new rent-paying accounts from being created by collecting fees into a non-existent validator id account
+
+#### `skip_rent_rewrites`
+Feature address: https://explorer.solana.com/address/CGB2jM8pwZkeeiXQ66kBMyBR6Np61mggL7XUsmLjVcrw
+Feature issue: https://github.com/solana-labs/solana/issues/26599
+Not yet activated on mainnet
+- When running partitioned rent collection for a block, only write the accounts
+that were actually changed (either rent epoch or rent amount) and only include
+those written accounts in the account delta hash
+- Note that this feature was modified in
+(https://github.com/anza-xyz/agave/pull/2910) to fix an issue where we would not
+rewrite account when we should have because the rent epoch was updated. This bug
+is actually not hit normally on a cluster like mainnet where all accounts
+already have their rent epoch set to the MAX marker value. (You would need to
+make a pre-existing rent-paying account into a rent exempt account and then rent
+collection would fail to set the rent epoch to the marker value
+`RENT_EXEMPT_RENT_EPOCH`). But there is an edge case (even on mainnet) when we
+create new sysvars, builtins, and precompiles because they get created with a
+default rent epoch of 0 and later get updated to MAX (though this changes
+slightly in `disable_rent_fees_collection`).
```

Comment:
@brooksprumo I wrote up this section while thinking about ways we could hit the bug on an existing cluster rather than just a new cluster as you found. Does my write up make sense to you?